packages/launcher/src/renderer/apps/AppContainer.js

```diff
@@ -4,21 +4,73 @@ import React, { Component } from 'react'
 import { View, StyleSheet, Text } from 'react-native-web'
 import path from 'path'
 import url from 'url'
+import type { ID } from '@mainframe/utils-id'
+
+import { callMainProcess } from '../electronIpc'
 
 declare var __static: string
 
-export default class App extends Component<{ appId: string }> {
+type User = {
+  id: ID,
+  data: Object,
+}
+
+type App = {
+  id: ID,
+  manifest: Object,
+}
+
+type Session = {
+  id: ID,
+  permission: Object,
+}
+
+type AppSessionData = {
+  app: App,
+  user: User,
+  session: Session,
+}
+
+type State = {
+  sessionData?: AppSessionData,
+}
+
+export default class AppContainer extends Component<
+  { windowId: string },
+  State,
+> {
+  state: State = {}
```

Comment:
Not an issue but no need to set the `State` type here, having it defined at the class level (`Component<Props, State>`) is enough to make Flow handle it.